### PR TITLE
Updating git ignore files to add test make files and eclipse IDE config files

### PR DESCRIPTION
**Pull request by RachelTucker**
Updating git ignore files to ensure various artifacts are not checked in.

### CODE REVIEW

File: test/.gitignore [hunk 1]
```diff
@@ -1 +1,9 @@
 test
+
+Makefile
+CTestTestfile.cmake
+ds3_c_tests
```
Comment (rpmoore) [inline test/.gitignore]: Is this a directory?  if so, the convention has been to put a / at the end of the directory name.

Comment (RachelTucker) [reply test/.gitignore]: ds3_c_tests is an executable

Comment (rpmoore) [reply test/.gitignore]: Got it

Comment (RachelTucker) [reply test/.gitignore]: I did update a separate entry that was a directory

**Pull request merged**